sound/soc/intel/boards/sof_sdw_common.h

```diff
@@ -12,6 +12,7 @@
 #include <linux/bits.h>
 #include <linux/types.h>
 #include <sound/soc.h>
+#include "../../codecs/rt711.h"
```

Comment:
add linux/soundwire/sdw.h above.

Reply:
> add linux/soundwire/sdw.h above.

I moved `#include "../../codecs/rt711.h"` to sof_sdw.c which is simpler. :) Somehow I didn't realize that `rt711.h` doesn't have to be in sof_sdw_common.h

Reply:
Thanks @plbossart and @bardliao :-) Basically the rules are quite simple and algorithmic (which makes them simple): you include headers in files, that need them. And to "need" a header file means to use a data type (including enums, structures and typedefs), a function declaration, a macro from that header. Note, that a pointer to a structure doesn't really need that header, it's sufficient (and preferred) to forward declare that structure. So if b.h defines
```
struct b_data {
	struct a_data a;
}
```
where `struct a_data` is defined in a.h, then b.h has to include a.h. Whereas for
```
struct b_data {
	struct a_data *a;
}
```
b.h can just contain a forward declaration `struct a_data;` usually right above the `struct b_data` declaration. Of course, if b.h anyway includes a.h for some other reason, then the forward declaration isn't needed.

Reply:
Thanks @lyakh for the explanation. It really helps. I have a question.
Forget about this PR and see [struct  rt711_priv](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/codecs/rt711.h#L13) in rt711.h. 

struct  rt711_priv {} uses	`enum sdw_slave_status` and `struct sdw_bus_params` which are defined in [linux/soundwire/sdw.h](https://github.com/thesofproject/linux/blob/topic/sof-dev/include/linux/soundwire/sdw.h). Does it mean that rt711.h should include <linux/soundwire/sdw.h>?
Or it is better to include <linux/soundwire/sdw.h> in each .c file which include "rt711.h"?

Reply:
@bardliao yes, it's per file. `enum sdw_slave_status` is used in the rt711.h header, so it has to be defined *there*. You can imagine an empty .c file, only including that single header - that should compile. And, something that wasn't clear to me, you cannot forward-declare enums (in ISO C according to https://stackoverflow.com/questions/7480217/c-forward-declaration-of-enums)

Reply:
> @bardliao yes, it's per file. `enum sdw_slave_status` is used in the rt711.h header, so it has to be defined _there_. You can imagine an empty .c file, only including that single header - that should compile. And, something that wasn't clear to me, you cannot forward-declare enums (in ISO C according to https://stackoverflow.com/questions/7480217/c-forward-declaration-of-enums)

Thanks @lyakh. Much appreciated.